applications/vanilla/controllers/class.postcontroller.php

```diff
@@ -162,6 +162,10 @@ public function discussion($CategoryUrlCode = '') {
             $this->title(t('New Discussion'));
         }
 
+        if (isset($this->Draft) && $this->Draft->Type) {
```

Comment:
Actually I'd just ditch this `if` entirely and just go with:

`$this->setData('Type', val('Type', $this->Draft, null));`